app/models/bundle_context.rb

```diff
@@ -21,4 +26,82 @@ class BundleContext < ApplicationRecord
     "smpl_style" => 2
   }
 
+  def staging_dir
+    '/dor/assembly'
+  end
+
+  def setup_bundle
+    self[:bundle_dir].chomp!("/") if bundle_dir
+  end
+
+  def progress_log_file
+    Tempfile.new.path(id) #FIXME: Figure out where to set this path or can remove. (planning meeting 9/10/18)
+  end
+
+  def content_exclusion #FIXME: Delete everywhere in code (#227)
+    nil
+  end
+
+  def file_attr
+    nil # FIXME can get rid of this (#228)
+  end
+
+  def validate_files?
+    "false" #FIXME delete everwhere in code (#230)
+  end
+
+  def content_tag_override? #TODO: find where this is used as a conditional and delete code that won't be executed (#231)
+    true
+  end
+
+  def smpl_manifest
+    'smpl_manifest.csv'
+  end
+
+  def manifest
+    'manifest.csv'
+  end
+
+  def path_in_bundle(rel_path)
+    File.join(bundle_dir, rel_path)
+  end
+
+  # TODO: BundleContext is not really a logical home for this util method
+  # load CSV allowing UTF-8 to pass through, deleting blank columns
+  # @param [String] filename
+  # @return [Array<ActiveSupport::HashWithIndifferentAccess>]
+  # @raise if file missing/unreadable
+  def self.import_csv(filename)
+    raise BundleUsageError, "CSV filename required" unless filename.present?
+    raise BundleUsageError, "Required file not found: #{filename}." unless File.readable?(filename)
+    file_contents = IO.read(filename).encode("utf-8", replace: nil)
+    csv = CSV.parse(file_contents, :headers => true)
+    csv.map { |row| row.to_hash.with_indifferent_access }
+  end
+
+  # On first call, loads the manifest data, caches results
+  # @return [Array<ActiveSupport::HashWithIndifferentAccess>]
+  def manifest_rows
+    @manifest_rows ||= self.class.import_csv(path_in_bundle(manifest))
+  end
+
+  def manifest_cols
+    {
+      label: 'label',
+      source_id: 'sourceid',
+      object_container: 'object', #object referring to filename or foldername
+      druid: 'druid'
+    }
+  end
+
+  private
+
+  def verify_bundle_directory
+    return if errors.key?(:bundle_dir)
+    errors.add(:bundle_dir, "Bundle directory: #{bundle_dir} not found.") unless File.directory?(bundle_dir)    
+  end
+
+  def verify_content_metadata_creation
+    errors.add(:content_metadata_creation, "The SMPL manifest #{smpl_manifest} was not found in #{bundle_dir}.") if smpl_style? && !File.exist?(File.join(bundle_dir, smpl_manifest)) 
```

Comment:
if i remove the guard clause in `verify_bundle_directory` i get an error 
```
 TypeError:
       no implicit conversion of nil into String
```

and that is because `bundle_dir` is nil. When i print the errors hash i do see 

```
#<ActiveModel::Errors:0x00007fab6c167028 @base=#<BundleContext id: nil, project_name: "SmokeTest", content_structure: "simple_book_structure", bundle_dir: "spec/test_data/bundle_input_g", staging_style_symlink: false, content_metadata_creation: nil, user_id: nil, created_at: nil, updated_at: nil>, @messages={:content_metadata_creation=>["can't be blank"]}, @details={:content_metadata_creation=>[{:error=>:blank}]}>
```

so to avoid the error, i added the guard clause.

Reply:
ah!  sure, sensible.  if the `bundle_dir` failed the preliminary validations, no reason to use it to check the file system.